code/game/atoms.dm

```diff
@@ -365,40 +365,143 @@
+
+
+//to add blood from a mob onto something, and transfer their dna info
+/atom/proc/add_mob_blood(mob/living/M)
+	var/list/blood_dna = M.get_blood_dna_list()
+	if(!blood_dna)
+		return 0
+	var/bloodcolor = "#A10808"
+	var/list/b_data = M.get_blood_data(M.get_blood_id())
+	if(b_data)
+		bloodcolor = b_data["blood_color"]
+
+	return add_blood(blood_dna, bloodcolor)
+
+//to add blood onto something, with blood dna info to include.
+/atom/proc/add_blood(list/blood_dna, color)
+	return 0
+
+/obj/add_blood(list/blood_dna, color)
+	return transfer_blood_dna(blood_dna)
+
+/obj/item/add_blood(list/blood_dna, color)
+	var/blood_count = !blood_DNA ? 0 : blood_DNA.len
+	if(!..())
+		return 0
+	if(!blood_count)//apply the blood-splatter overlay if it isn't already in there
+		add_blood_overlay(color)
+	return 1 //we applied blood to the item
+
+/obj/item/clothing/gloves/add_blood(list/blood_dna, color)
+	. = ..()
+	transfer_blood = rand(2, 4)
+
+/turf/add_blood(list/blood_dna, color)
+	var/obj/effect/decal/cleanable/blood/splatter/B = locate() in src
+	if(!B)
+		B = new /obj/effect/decal/cleanable/blood/splatter(src)
+	B.transfer_blood_dna(blood_dna) //give blood info to the blood decal.
+	B.basecolor = color
+	return 1 //we bloodied the floor
+
+/mob/living/carbon/human/add_blood(list/blood_dna, color)
+	if(wear_suit)
+		wear_suit.add_blood(blood_dna, color)
+		wear_suit.blood_color = color
+		update_inv_wear_suit(1)
+	else if(w_uniform)
+		w_uniform.add_blood(blood_dna, color)
+		w_uniform.blood_color = color
+		update_inv_w_uniform(1)
+	if(gloves)
+		var/obj/item/clothing/gloves/G = gloves
+		G.add_blood(blood_dna, color)
+		G.blood_color = color
+	else
+		hand_blood_color = color
+		bloody_hands = rand(2, 4)
+		transfer_blood_dna(blood_dna)
+
+	update_inv_gloves(1)	//handles bloody hands overlays and updating
 	return 1
 
-// Only adds blood on the floor -- Skie
-/atom/proc/add_blood_floor(mob/living/carbon/M)
-	return //why the fuck this is at an atom level but only works on simulated turfs I don't know
+/obj/item/proc/add_blood_overlay(color)
+	if(blood_overlay)
+		return
+	if(initial(icon) && initial(icon_state))
+		//try to find a pre-processed blood-splatter. otherwise, make a new one
+		var/index = blood_splatter_index()
+		var/icon/blood_splatter_icon = blood_splatter_icons[index]
+		if(!blood_splatter_icon)
+			blood_splatter_icon = icon(initial(icon), initial(icon_state), , 1)		//we only want to apply blood-splatters to the initial icon_state for each object
+			blood_splatter_icon.Blend("#fff", ICON_ADD) 			//fills the icon_state with white (except where it's transparent)
+			blood_splatter_icon.Blend(icon('icons/effects/blood.dmi', "itemblood"), ICON_MULTIPLY) //adds blood and the remaining white areas become transparant
+			blood_splatter_icon = fcopy_rsc(blood_splatter_icon)
+			blood_splatter_icons[index] = blood_splatter_icon
+
+			blood_overlay = image(blood_splatter_icon)
+			blood_overlay.color = color
+			overlays += blood_overlay

```

Comment:
Couldn't the above three lines just be:

```
blood_spatter_icon.color = color
overlays += blood_splatter_icon
```

and indent it one to the left?

otherwise, you're not going to get the overlay added if the overlays already exists.